### PR TITLE
webui: improve item param description

### PR DESCRIPTION
**Pull request by or2e**
Made it more flexible to manage the extended description of item parameters

<details>
  <summary>Filter</summary>

  image
</details>

<details>
  <summary>Item detail</summary>

  image
</details>

### CODE REVIEW

File: src/WebUI/locales/en.yml [hunk 1]
```diff
@@ -1019,12 +1019,19 @@ item:
     Cut:
       long: 'Cut'
       short: 'c'
+      title: 'Cut damage'
+      description: 'Cut damage description'
     Pierce:
       long: 'Pierce'
       short: 'p'
+      title: 'Pierce damage'
+      description: 'Pierce damage description'
```
Comment (Yeldur) [inline src/WebUI/locales/en.yml:1028]: Pierce damage, on the other hand, signifies the damage dealt by weapons designed to puncture or penetrate, such as arrows, spears, or daggers. The key strength of pierce damage lies in its efficiency against armoured opponents. Pierce damage offers a headshot multiplier increase greater than usual, if a melee (pierce) attack lands on an enemy's head, the damage increases by 1.3 times, while for ranged attacks, the increment is 1.6 times. This makes pierce damage the strongest potential damage type given its benefit from headshots.

File: src/WebUI/locales/en.yml [hunk 2]
```diff
@@ -1019,12 +1019,19 @@ item:
...
     Blunt:
       long: 'Blunt'
       short: 'b'
+      title: 'Blunt damage'
+      description: 'Blunt damage description'
```
Comment (Yeldur) [inline src/WebUI/locales/en.yml:1033]: Blunt damage refers to the trauma caused by impact weapons like hammers, maces, or clubs. This type of damage is akin to pierce damage in terms of its effectiveness against armoured foes. However, unlike pierce damage, blunt damage does not have an increased headshot multiplier. This makes blunt better for general use comparatively to that of pierce.

File: src/WebUI/src/pages/shop.vue [hunk 3]
```diff
@@ -193,6 +193,7 @@ const newItemCount = computed(
       sticky-header
       :detailed="isUpgradableCategory"
       detailKey="id"
+      customRowKey="id"
```
Comment (or2e) [inline src/WebUI/src/pages/shop.vue:196]: this is an important, fixes broken `Tooltip` when changing the filter by item type

File: src/WebUI/locales/ru.yml [hunk 4]
```diff
@@ -38,24 +38,24 @@ dateTime:
...
-  yesterday: "вчера"
-  tomorrow: "завтра"
-  day: "дней | день | дня | дней"
-  last-week: "на прошлой неделе"
-  next-week: "на следующей неделе"
-  week: "недель | неделя | недели | недель"
-  hour: "часов | час | часа | часов"
-  minute: "минут | минуты | минуты | минут"
-  second: "секунд | секунда | секунды | секунд"
+  just-now: 'прямо сейчас'
```
Comment (or2e) [inline src/WebUI/locales/ru.yml:41]: oops, I apologize, Prettier was broken when I did this.

File: src/WebUI/src/components/item/ItemFieldIcon.vue [hunk 5]
```diff
@@ -1,38 +1,20 @@
...
   icon: IconedBucket;
-  label: string;
-  description?: string | null;
   size?: 'xl' | '2xl';
-  showTooltip?: boolean;
 }>();
 </script>
 
 <template>
-  <Tooltip
```
Comment (or2e) [inline src/WebUI/src/components/item/ItemFieldIcon.vue:20]: Tooltip now wraps around in elsewhere

File: src/WebUI/src/components/item/ItemParam.vue [hunk 6]
```diff
@@ -89,29 +89,25 @@ const fieldStyle = computed(() => {
...
-
-      <template v-else>
-        <ItemFieldIcon
-          v-if="formattedValue.icon !== null"
-          :icon="formattedValue.icon"
-          :label="(formattedValue.label as string)"
-          size="2xl"
-        />
-
+      <slot v-bind="{ rawBuckets, formattedValue, diffStr }">
```
Comment (or2e) [inline src/WebUI/src/components/item/ItemParam.vue:92]: simplified the cmp

File: src/WebUI/src/components/shop/ShopGridFilter.vue [hunk 7]
```diff
@@ -41,7 +41,7 @@ const props = defineProps<{
         </div>
       </Tooltip>
 
-      <template #popper>
+      <template #popper="{ hide }">
```
Comment (or2e) [inline src/WebUI/src/components/shop/ShopGridFilter.vue:44]: Close after selection

File: src/WebUI/src/services/item-service.ts
Comment (or2e) [inline src/WebUI/src/services/item-service.ts:1]: a lot of changes, it's because of the change to a more user-friendly data structure.

**Review by verdie-g — APPROVED** [review]

**Pull request merged**